src/ArraySetopsMsg.chpl

```diff
@@ -53,28 +53,34 @@ module ArraySetopsMsg
         var intersect_maxMem = max(gEnt_sortMem, gEnt2_sortMem);
         overMemLimit(intersect_maxMem);
 
-        select(gEnt.dtype) {
-          when (DType.Int64) {
-             if (gEnt.dtype != gEnt2.dtype) {
-                 var errorMsg = notImplementedError("newIntersect1d",gEnt2.dtype);
-                 asLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);                             
-                 return new MsgTuple(errorMsg, MsgType.ERROR);
-             }
-             var e = toSymEntry(gEnt,int);
-             var f = toSymEntry(gEnt2, int);
-             
-             var aV = intersect1d(e.a, f.a, isUnique);
-             st.addEntry(vname, new shared SymEntry(aV));
+        select(gEnt.dtype, gEnt2.dtype) {
+          when (DType.Int64, DType.Int64) {
+            var e = toSymEntry(gEnt,int);
+            var f = toSymEntry(gEnt2,int);
 
-             repMsg = "created " + st.attrib(vname);
-             asLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),repMsg);
-             return new MsgTuple(repMsg, MsgType.NORMAL);
-           }
-           otherwise {
-               var errorMsg = notImplementedError("newIntersect1d",gEnt.dtype);
-               asLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);           
-               return new MsgTuple(errorMsg, MsgType.ERROR);
-           }
+            var aV = intersect1d(e.a, f.a, isUnique);
+            st.addEntry(vname, new shared SymEntry(aV));
+
+            repMsg = "created " + st.attrib(vname);
+            asLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),repMsg);
+            return new MsgTuple(repMsg, MsgType.NORMAL);
+          }
+          when (DType.UInt64, DType.UInt64) {
+            var e = toSymEntry(gEnt,uint);
+            var f = toSymEntry(gEnt2,uint);
+
+            var aV = intersect1d(e.a, f.a, isUnique);
+            st.addEntry(vname, new shared SymEntry(aV));
+
+            repMsg = "created " + st.attrib(vname);
+            asLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),repMsg);
+            return new MsgTuple(repMsg, MsgType.NORMAL);
+          }
+          otherwise {
+            var errorMsg = notImplementedError("newIntersect1d",gEnt.dtype);
```

Comment:
is the string here intended? `"newIntersect1d"`

Reply:
I think that string was present before this PR

Reply:
Ya, that string does predate this PR. I thought it was weird and just kept it at first, but now that I think about it a bit more, when I was an intern and implemented this intersect code I had 2 versions of intersect around and called this new one `newIntersect1d`, so I think that just slipped past my intern project... I will fix that here.